File: node-service/src/repositories/sensorRepository.js
class SensorRepository {
  constructor(storage) {
    this.storage = storage;
  }

  findAll() {
    return Array.from(this.storage.values());
  }

  findById(id) {
    return this.storage.get(id) || null;
  }

  create(sensorData) {
    const id = sensorData.sensor_id;
    this.storage.set(id, sensorData);
    return sensorData;
  }

  exists(id) {
    return this.storage.has(id);
  }
}

module.exports = SensorRepository;

File: python-service/src/repositories/sensor_repository.py
from typing import Dict, List, Optional, Any


class SensorRepository:
    def __init__(self, storage: Dict[str, Dict[str, Any]]):
        self._storage = storage

    def find_all(self) -> List[Dict[str, Any]]:
        return list(self._storage.values())

    def find_by_id(self, sensor_id: str) -> Optional[Dict[str, Any]]:
        return self._storage.get(sensor_id)

    def create(self, sensor_data: Dict[str, Any]) -> Dict[str, Any]:
        sensor_id = sensor_data['sensor_id']
        self._storage[sensor_id] = sensor_data
        return sensor_data

    def exists(self, sensor_id: str) -> bool:
        return sensor_id in self._storage
